### DISCUSSION TITLE
add data in requiredIf closure parameter

### DISCUSSION THREAD

**Discussion by comhon-project** [laravel/framework#44985] [Ideas]
Hello, I would like to do a request feature for `Rule::requiredIf` closure.
Actual : 
`requiredIf` closure doesn't have parameter

Expected : 
`requiredIf` closure contains all data to be validated as first parameter.

example : 
```PHP
Validator::make($inputs, [
    'my_property' => Rule::requiredIf(function ($inputs) {
        // check whatever in $inputs
    ),
]);
```
Obviously in this simple example I may access $inputs like this 
```PHP
Rule::requiredIf(function () use ($inputs) {...})
```
but in a more complex situation where i would not have access to `$inputs` It would be very helpfull to have this parameter.
(in my case I have dynamic models and my rules are built dynamically and I can't access `$inputs`).

is it something possible and desirable ?

**Comment by henzeb** [comment]
As long it is not available, this package may be a nice workaround. https://packagist.org/packages/henzeb/laravel-console-facade

**Reply by comhon-project** [reply]
thanks for your answer but I don't understand how your library can help me oO
